### PR TITLE
KB-H058 Disallowed characters

### PR DESCRIPTION
**Pull request by uilianries**
Hello! This PR implements #298 

Any reserved character won't be allowed anymore. The error message will list the file name and ask for renaming.

closes https://github.com/conan-io/hooks/issues/298

/cc @madebr

### CODE REVIEW

**Review by madebr — APPROVED** [review]
Looks good to me!

File: hooks/conan-center.py [hunk 1]
```diff
@@ -609,6 +610,16 @@ def test(out):
                 out.error("Both '{0}requires' attribute and '{0}requirements()' method should not "
                           "be declared at same recipe.".format(prefix))
 
+    @run_test("KB-H058", output)
+    def test(out):
+        disallowed_chars = '<>:"/\\|?*'
```
Comment (jgsogo) [inline hooks/conan-center.py:615]: Even if they are not strictly forbidden I would add some more rules in this hook. Most taken from here (https://en.wikipedia.org/wiki/Filename#Reserved_characters_and_words):
 * characters `,`, `;`, `%`
 * blank/spaces
 * `.` at the end

Can you figure out any reason to have those characters in file/folder names?

**Review by jgsogo — COMMENTED** [review]
I like it, preventing weird filenames is a condition to sanity 😄

File: hooks/conan-center.py (hunk 1 above)
Comment (uilianries) [reply hooks/conan-center.py:615]: patch files and extra testing files (e.g xml), people can use any name they "want".

Comment (uilianries) [reply hooks/conan-center.py]: Done

**Review by jgsogo — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**